File: backend/app/domain/models/notification.py
from pydantic import BaseModel
from datetime import datetime
from enum import Enum


class NotificationType(Enum):
    INFO = "info"
    WARNING = "warning"
    CRITICAL = "critical"
    ERROR = "error"


class Notification(BaseModel):
    id: int | None = None
    message: str
    read_status: bool = False
    type: NotificationType
    title: str
    device_id: int | None = None
    ts: datetime = datetime.now()

    class Config:
        use_enum_values = True

File: frontend/app/src/main/java/com/example/iot/data/viewmodels/Notification.kt
package com.example.iot.data.viewmodels

import com.example.iot.R

data class Notification(
    var id: Int,
    var message: String,
    var read_status: Boolean,
    var type: String,
    var title: String,
    var device_id: Int,
    var ts: String,
    val iconRes: Int = R.drawable.ic_notifications
) 